### PR TITLE
Introduce asynchronous rendering

### PR DESCRIPTION
**Pull request by devballteam**

### CODE REVIEW

File: decoupled/component.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ module.exports = (function () {
     var repository = Component._repository.get(component.name);
 
     repository.find(component.type, function (data) {
-      done( Component._renderer.render(component.templatePath, data) );
+      Component._renderer.render(component.templatePath, data, done);
```
Comment (marekbogatzki) [inline decoupled/component.js]: Maybe you could extend it instead changing. I mean maybe create method `renderAsyncand` give possibility to choose how you want to render async or sync. What do you think?

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+cat decoupled.js decoupled/* > decoupled-browser.js
```
Comment (marekbogatzki) [inline build.sh]: Why it's not in `package.json`?

File: decoupled/component.js (hunk 1 above)
Comment (devballteam) [reply decoupled/component.js]: Thats a good idea. We can use ``setup`` method for it

File: build.sh (hunk 2 above)
Comment (devballteam) [reply build.sh]: It can be. I just did it on the fly in my way. I will change that.

File: module-polyfill.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+!function(e){if("object"==typeof exports&&"undefined"!=typeof module)module.exports=e();else if("function"==typeof define&&define.amd)define([],e);else{var f;"undefined"!=typeof window?f=window:"undefined"!=typeof global?f=global:"undefined"!=typeof self&&(f=self),f.DC=e()}}(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
+
```
Comment (esuszek) [inline module-polyfill.js]: Why this polyfill is finished with ``[function(require,module,exports){`` ?

Comment (devballteam) [reply module-polyfill.js]: I don't know. Its polyfill of ``module, exporta, import`` (something like that) feature that is not supported yet in any browser. I created this file and commited it by mistake. Removed.

File: build.sh (hunk 2 above)
Comment (devballteam) [reply build.sh]: Done

File: decoupled/component.js (hunk 1 above)
Comment (devballteam) [reply decoupled/component.js]: Done

**Review by marekbogatzki — APPROVED** [review]

**Review by tomaszharazin — APPROVED** [review]

**Review by mateuszlisperform — APPROVED** [review]

**Review by esuszek — APPROVED** [review]

**Comment by mpietrowiak** [comment]
.

**Review by mpietrowiak — APPROVED** [review]

**Pull request merged**